### DISCUSSION TITLE
New copy icons in the food log

### DISCUSSION THREAD

**Discussion by pjfarr** [jlai/fitness-dashboard#59] [Ideas]
Hi Jason!

This morning when using the food log I noticed the addition of individual "copy" icons beside each amount in the Daily Total row when hovered over. I'm not sure what purpose they serve as clicking them does not transfer the numbers to the clipboard. However, they do interfere with the ability to select and copy the entire row at once, which is something I do every day to transfer the daily totals to a spreadsheet. Is it possible to undo this addition? Or is there a practical purpose for it I'm not seeing?

You're doing a great job, really like this dashboard more than the original Fitbit site. So grateful you've done this!

**Comment by jlai** [comment]
Hi PJ, this feature was requested (and contributed) by @nicolas-girod . I think we'll probably add a setting to disable this (pull request #61).

We're also thinking about having a button to quick copy rows or the entire table to the clipboard. Do you just want to copy the daily totals, or do you copy the meal times or individual food items? Would it make sense to have a .csv export (e.g. for a week at a time)?

**Reply by pjfarr** [reply]
Options are always good! I copy the daily total line for calories/macros etc. to paste into a spreadsheet. When pasting, the copied data automatically separates into individual cells, which I like. A .csv export of the daily food log a week at a time would be great. Fitbit used to offer this as a .csv download along with other activity stats pre-Google.
Thanks Jason, I appreciate your response and your commitment!

EDIT: I see the icons no longer pop up! If you adjusted the code, thanks!

**Reply by nicolas-girod** [reply]
Might be hard to offer a weekly export because the website doesn't have any backend and calls into the Fitbit API directly. Exporting the content of a day from the datagrid might be the best we can offer.
EDIT: Might be doable with this [endpoint](https://dev.fitbit.com/build/reference/web-api/nutrition-timeseries/get-nutrition-timeseries-by-date-range/).

**Reply by pjfarr** [reply]
Hey, the export feature would just be icing on an already great project! No complaints here, I'm just grateful for Jason's initiative in making this available and that he's open and responsive to suggestions. In a very short time his dashboard went so above and beyond the original Fitbit site that I'd be hard pressed to return to the Fitbit version if they ever brought it back (fat chance).

I notice you're designated as "Collaborator". Does that mean you're contributing code as well? Assisting in the design? If so, I'd like to thank you too. I know you guys are doing this on your own time,

**Reply by nicolas-girod** [reply]
I have contributed some features yes. As Jason pointed out, that copy button was from me (sorry it caused you issues). I agree that the dashboard is even better than before (I love being able to copy food from one day to the next). I admit I've been pushing stuff that I'm interested in having as a feature but I recognise it might not match with everyone's needs/tastes. I hope Jason doesn't mind too much my messing up his code base 😅 You should raise an [issue](https://github.com/jlai/fitness-dashboard/issues) for the export feature and one of us will get to it eventually.

**Comment by nicolas-girod** [comment]
Choice is always good so +1 for the option. Out of curiosity, which browser are you using? Copying is working fine for me in Brave on both mobile and desktop.

**Reply by pjfarr** [reply]
I'm using Chrome, but I also tried it on Firefox and Edge. The little copy icons pop up to the left of each column header when hovered over. It prevents me from drag-selecting the numbers. Clicking the icons doesn't seem to do anything (nothing transferred to clipboard). Even if they worked, I wouldn't have a use for copying each number individually. An icon that would copy the entire row would be good.

EDIT: I noticed yesterday the icons no longer pop up. I'm not sure if Jason adjusted the code or if it was a glitch on my end. Whatever the reason, glad they're gone!

**Reply by nicolas-girod** [reply]
As he said, he added an option to toggle them on/off. There will be a new option in the settings for that (off by default). I tried Firefox and it also works for me ¯\_(ツ)_/¯. Copying the whole line at once should be doable and Jason suggested it but if it doesn't work for individual value for you, it might not work still for the whole row. I also copy paste into a spreadsheet but it doesn't match the order of the values in the dashboard (that's why options are good :)
![image](https://github.com/user-attachments/assets/a85a8c2d-bee3-4320-8f42-6e4dc7b3bc8f)
